### DISCUSSION TITLE
Qodana Cloud - Compliance Issue

### DISCUSSION THREAD

**Discussion by aspexdaniel** [JetBrains/Qodana#195] [Q&A]
It seems the newest Github Action(v2023.2 ) now requires a Qodana Cloud token, however we're under compliance restriction so would like to know why it changes to mandatory and what data will be transferred to the cloud, thanks.

**Comment by brichbash** [comment]
Hello,

[Qodana Cloud](https://www.jetbrains.com/help/qodana/cloud-about.html) is required for our licensing logic (we need to count active contributors in your repositories, as our pricing depends on this number), and it's also used to store your reports. The analysis will still be executed in your pipelines, so your code won't leave your pipelines aside from little code snippets necessary for showing problems in reports.

We are working on an on-premises solution. It is planned for mid-2024, and it will have different pricing. Please contact us at qodana-sales@jetbrains.com to discuss your case in detail.
